rs/sns/governance/api/src/ic_sns_governance.pb.v1.rs

```diff
@@ -480,6 +480,36 @@ pub struct RegisterDappCanisters {
     /// At least one canister ID is required.
     pub canister_ids: Vec<::ic_base_types::PrincipalId>,
 }
+
+#[derive(
+    candid::CandidType, candid::Deserialize, comparable::Comparable, Clone, Debug, PartialEq,
+)]
+pub enum PreciseValue {
```

Comment:
I'm still a bit on the fence about whether or not this is actually going to be a good design decision.  It occurs to me we might want to support Principal as a type though.

Reply:
Yes, I was thinking about whether to add principal. 

The original type was from ICRC3, but they didn't even have `bool`. I think they just minimized it for their use case. 

I think I'm going to add principal.

Reply:
OTOH, principals can be easily encoded as strings, so it's not a show stopper if we don't include them, just a slight inconvenience to know the right encoding (bit at least there's a canonical string encoding for principals).